parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.java

```diff
@@ -87,7 +87,6 @@ protected ParselyTracker(String siteId, int flushInterval, Context c) {
         this.rootUrl = "https://p1.parsely.com/";
         this.queueSizeLimit = 50;
         this.storageSizeLimit = 100;
-        this.deviceInfo = this.collectDeviceInfo();
```

Comment:
Good point with keeping responsibilities seperately but `collectDeviceInfo` uses `adKey` property, which is assigned in `GetAdKey`, so that's why I moved creation of `deviceInfo` to `GetAdKey`.